Migrate Evince to Gnome-42/Core-22

sergio-costas opened the pull request:
This Snap also compiles libpoppler to allow to have it updated,
but compiling it with libnss3 fails.

# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please check only the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**Test Configuration**:

* OS (please include version):
* Any other relevant environment information:

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings

sergio-costas committed:
Migrate Evince to Gnome-42/Core-22

This Snap also compiles libpoppler to allow to have it updated,
but compiling it with libnss3 fails.

sergio-costas committed:
Added parse-info

sergio-costas committed:
migrate to the new CUPS

sergio-costas committed:
Removed libpoppler

It is now in the SDK.

hellsworth commented on snapcraft.yaml:
```diff
@@ -7,7 +7,20 @@ description: |
+assumes: [snapd2.55]
+
+# this is not used or needed for anything other than to trigger automatic
+# installation of the cups snap via "default-provider: cups"
```
Will every snap that needs cups in the future need this `foo-install-cups` plugs section?

sergio-costas replied:
I added that because the "migration guide" (https://forum.snapcraft.io/t/new-interface-cups-for-all-snaps-which-print/29702) stated so. But I asked in the group if it is still needed.

sergio-costas replied:
I asked, and they said that it is still needed.

hellsworth commented on snapcraft.yaml:
```diff
@@ -19,44 +32,57 @@ slots:
   evince-thumbnailer:
     command: usr/bin/evince-thumbnailer
     plugs:
       - gsettings
```
gsettings is part of the gnome extension, so it can be removed.

sergio-costas replied:
But are you talking about the library, or the plug? Because, AFAIU, this is to give the snapped program access to the user's settings... does using the Gnome extension give that access automagically?

hellsworth replied:
I'm talking about the plug, at least [for gnome-3-38](https://github.com/snapcore/snapcraft/blob/main/snapcraft_legacy/internal/project_loader/_extensions/gnome_3_38.py#L117) this was the case.

kenvandine replied:
The plug is added automatically when using the gnome extension.  You can verify by removing it in the yaml then running "snapcraft expand-extensions" and you'll see gsettings plug is still in the resulting yaml.

sergio-costas replied:
It complains about the "parse-info" statement, but if I remove it, you are right: gsettings is there. Removing.

sergio-costas replied:
Done.

hellsworth commented on snapcraft.yaml:
```diff
@@ -19,44 +32,57 @@ slots:
+    source-tag: '42.3'
     plugin: meson
+    parse-info: [usr/share/metainfo/org.gnome.Evince.appdata.xml]
+    build-environment: *buildenv
```
since the evince part is the only one that uses the `PKG_CONFIG_PATH` set in the buildenv part, just define it here instead and remove that buildenv part.

sergio-costas replied:
Good point. Done.

sergio-costas committed:
Move environment into part

There was an environment definition which was used only by one
part. This patch removes it and moves the definition into the
part.

sergio-costas committed:
Remove gsettings plug

The gsettings plug is already set automagically when using the
gnome extension, so it's not needed.

hellsworth commented:
looks great thanks!
